plugins/modules/content_upload.py

```diff
@@ -157,6 +175,12 @@ def main():
         elif module.foreman_params['repository']['content_type'] == 'file':
             query = 'name = "{0}" and checksum = "{1}"'.format(filename, checksum)
             content_unit = module.find_resource('file_units', query, params=repository_scope, failsafe=True)
+        elif module.foreman_params['repository']['content_type'] == 'ostree':
+            query = 'name = "{0}" and checksum = "{1}"'.format(filename, checksum)
```

Comment:
So, as discussed yesterday on IRC, this is what "fails" your idempotency test.

1. When I execute this search manually, the API tells me `Field 'checksum' not recognized for searching!` and is not finding anything.
2. Looking at the tarball you're using for this test, it has two "heads": `rawhide` and `stable`, so when I just list all known ostree refs, I get: 
    ```json
    "results":[
      {"id":2,"name":"rawhide","filename":null,"pulp_id":"/pulp/api/v3/content/ostree/refs/9ae5bf3d-525d-4a27-a527-d08f85f62786/","version":null,"content_type":"ostree_ref","additional_metadata":null},
      {"id":1,"name":"stable","filename":null,"pulp_id":"/pulp/api/v3/content/ostree/refs/9cf81424-9e75-4e15-b31f-a03d607d5890/","version":null,"content_type":"ostree_ref","additional_metadata":null}
    ]
    ```
    thus, searching for `name = <filename>` doesn't yield anything either.
3. We could consider using the `filename` field for searching, but this is `null` for this upload?

Reply:
IIRC the checksum reported in Katello for those units will be the checksum of the OStree commit, not the checksum of the file itself.

I'll see about an alternate search field.